ComponentKit/Core/CKComponent.mm

```diff
@@ -166,10 +166,11 @@ - (void)unmount
 {
   CKAssertMainThread();
   if (_mountInfo != nullptr) {
-    [_scopeHandle.controller componentWillUnmount:self];
+    CKComponentController *const controller = (CKComponentController *)_scopeHandle.controller;
```

Comment:
Is there a way we can use Objective-C generics on the scope handle to avoid these casts?